### PR TITLE
Armory - Minor fixes and some little features

### PR DESCRIPTION
**Pull request by PhILoXs**
**When merged this pull request will:**
- Fix deletion of loadouts and backbacks not propagating to all clients and set default values (nil ist reserved)
- Check if loadout or backback are available and if not print a warning to rpt
- Add more logging in debug mode
- Add setting to connect as player to the database (For mission testing)
- Add ACE Interaction for new equipment creation, if available.
- Improve createEquipment function

### CODE REVIEW

File: addons/armory/functions/fnc_createEquipment.sqf [hunk 1]
```diff
@@ -8,33 +8,31 @@
...
 };
 
-CHECK(!GVAR(initialized) || !_params || _equipmentName isEqualTo "" || ((_editors param [ARR_2(0,"")]) isEqualTo "") || !SERVER_CHECK);
+CHECK(!GVAR(initialized) || _equipmentName isEqualTo "" || ((_editors param [ARR_2(0,"")]) isEqualTo ""));
+CHECK(!isDedicated && !GVAR(cba_settings_playerDBConnection));
 
 [GVAR(sessionID), "insertEquipment", _equipmentName, _editors] call DB_SET;
 
-if (_initialize) then {
+if !(isNull _object) then {
```
Comment (Timi007) [inline addons/armory/functions/fnc_createEquipment.sqf:33]:
```suggestion
if (!isNull _object) then {
```

File: addons/armory/functions/fnc_equipBackpack.sqf [hunk 2]
```diff
@@ -25,13 +25,16 @@ CHECK(!GVAR(initialized) || !_params);
...
-    LOG("Equipment not in Namespace");
+    LOG_1("Equipment '%1' not in namespace. Will be initialized.",_equipmentName);
     [QGVAR(initEquipment), [player, QGVAR(equipBackpack), [_equipmentName, _backpackName]]] call CBA_fnc_serverEvent;
 };
 
 _equipmentNamespaces params ["", "_backpackNamespace"];
 
-(_backpackNamespace getVariable [_backpackName, []]) params ["", "", "_class", "_items"];
+private _backpack = _backpackNamespace getVariable [_backpackName, []];
+CHECKRET(_backpack isEqualTo [], WARNING_1(QUOTE(Backpack '%1' not available!), _backpackName));
```
Comment (Timi007) [inline addons/armory/functions/fnc_equipBackpack.sqf:35]:
```suggestion
CHECKRET(_backpack isEqualTo [], WARNING_1("Backpack '%1' not available!", _backpackName));
```

File: addons/armory/functions/fnc_equipLoadout.sqf [hunk 3]
```diff
@@ -25,13 +25,16 @@ CHECK(!GVAR(initialized) || !_params);
...
-    LOG("Equipment not in Namespace");
+    LOG_1("Equipment '%1' not in namespace. Will be initialized.",_equipmentName);
     [QGVAR(initEquipment), [player, QGVAR(equipLoadout), [_equipmentName, _loadoutName]]] call CBA_fnc_serverEvent;
 };
 
 _equipmentNamespaces params ["_loadoutNamespace"];
 
-(_loadoutNamespace getVariable [_loadoutName, []]) params ["", "", "_loadout", "_ace_medic", "_ace_engineer"];
+private _loadout = _loadoutNamespace getVariable [_loadoutName, []];
+CHECKRET(_loadout isEqualTo [], WARNING_1(QUOTE(Loadout '%1' not available!), _loadoutName));
```
Comment (Timi007) [inline addons/armory/functions/fnc_equipLoadout.sqf:35]:
```suggestion
CHECKRET(_loadout isEqualTo [], WARNING_1("Loadout '%1' not available!", _loadoutName));
```

File: addons/armory/stringtable.xml [hunk 4]
```diff
@@ -1,6 +1,12 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project name="mts">
 	<Package name="Armory">
+        <Container name="misc">
+            <Key ID="STR_mts_Armory_subCategory">
+                <Original>Armory</Original>
+                <German>Armory</German>
+            </Key>
+        </Container>
```
Comment (Timi007) [inline addons/armory/stringtable.xml:9]: Obsolete. There's already a `STR_mts_Armory_displayName` entry with same text

File: addons/armory/stringtable.xml [hunk 5]
```diff
@@ -46,8 +52,8 @@
             <German>Erstellen</German>
         </Key>
         <Key ID="STR_mts_Armory_checkbox_initializeEquipmentCategory_tooltip">
-            <Original>Initializes equipment category after creation and if possible adds action for everyone</Original>
-            <German>Initialisiert die Ausrüstungskategorie und fügt, falls möglich, eine Aktion für jeden hinzu</German>
+            <Original>If possible, adds interaction menu for the newly created equipment.\n(Depends if an object was given to the openArmoy function)</Original>
+            <German>Wenn möglich, fügt ein Interaktionsmenü für die neu erstellte Ausrüstungskategorie hinzu.\n(Abhängig davon, ob ein Objekt an die Funktion openArmoy übergeben wurde)</German>
```
Comment (Timi007) [inline addons/armory/stringtable.xml:56]: Not sure if `\n` is fine. Normally `<br />` is used.

```suggestion
            <Original>If possible, adds interaction menu for the newly created equipment.<br />(Depends if an object was given to the openArmoy function)</Original>
            <German>Wenn möglich, fügt ein Interaktionsmenü für die neu erstellte Ausrüstungskategorie hinzu.<br />(Abhängig davon, ob ein Objekt an die Funktion openArmoy übergeben wurde)</German>
```

File: addons/armory/stringtable.xml [hunk 6]
```diff
@@ -46,8 +52,8 @@
             <German>Erstellen</German>
         </Key>
         <Key ID="STR_mts_Armory_checkbox_initializeEquipmentCategory_tooltip">
-            <Original>Initializes equipment category after creation and if possible adds action for everyone</Original>
-            <German>Initialisiert die Ausrüstungskategorie und fügt, falls möglich, eine Aktion für jeden hinzu</German>
+            <Original>If possible, adds interaction menu for the newly created equipment.\n(Depends if an object was given to the openArmoy function)</Original>
```
Comment (Timi007) [inline addons/armory/stringtable.xml:55]:
```suggestion
            <Original>If possible, adds interaction menu for the newly created equipment.\n(Depends on if an object was passed to the openArmoy function)</Original>
```

File: addons/armory/ui/RscAttributes.hpp [hunk 7]
```diff
@@ -62,12 +62,7 @@ class GVAR(dialog) {
             y = "SafeZoneY + (180 / 1080) * SafeZoneH";
             w = "(30 / 1920) * SafeZoneW";
             h = "(30 / 1080) * SafeZoneH";
-            onButtonClick = QUOTE(\
-                private _object = ((_this select 0) getVariable [ARR_2(QQGVAR(attachedObject),objNull)]);\
-                closeDialog 0;\
-                createDialog QQGVAR(newEquipment);\
-                ((findDisplay IDD_NEW_EQUIPMENT) displayCtrl IDC_BUTTON_CREATE) setVariable [ARR_2(QQGVAR(attachedObject), _object)];\
-            );
+            onButtonClick = QUOTE(private _object = ((_this select 0) getVariable [ARR_2(QQGVAR(attachedObject),objNull)]);closeDialog 0;createDialog QQGVAR(newEquipment);((findDisplay IDD_NEW_EQUIPMENT) displayCtrl IDC_BUTTON_CREATE) setVariable [ARR_2(QQGVAR(attachedObject), _object)];);
```
Comment (Timi007) [inline addons/armory/ui/RscAttributes.hpp:65]: This should be a function.

File: addons/armory/ui/RscAttributes.hpp [hunk 8]
```diff
@@ -1081,7 +1076,7 @@ class GVAR(newEquipment) {
             y = "SafeZoneY + (525 / 1080) * SafeZoneH";
             w = "(150 / 1920) * SafeZoneW";
             h = "(30 / 1080) * SafeZoneH";
-            onButtonClick = QUOTE([ARR_2(QQGVAR(createEquipment), [ARR_4(ctrlText IDC_EDITBOX_EQUIPMENT, [(getPlayerUID player)], cbChecked ((ctrlParent (_this select 0)) displayCtrl IDC_CHECKBOX_EQUIPMENT), (_this select 0) getVariable [ARR_2(QQGVAR(attachedObject), objNull)])])] call CBA_fnc_serverEvent; closeDialog 0);
+            onButtonClick = QUOTE(if (cbChecked ((ctrlParent (_this select 0)) displayCtrl IDC_CHECKBOX_EQUIPMENT)) then {[ARR_2(QQGVAR(createEquipment), [ARR_3(ctrlText IDC_EDITBOX_EQUIPMENT, [(getPlayerUID player)], (_this select 0) getVariable [ARR_2(QQGVAR(attachedObject), objNull)])])] call CBA_fnc_serverEvent;} else {[ARR_2(QQGVAR(createEquipment), [ARR_2(ctrlText IDC_EDITBOX_EQUIPMENT, [(getPlayerUID player)])])] call CBA_fnc_serverEvent;};closeDialog 0;);
```
Comment (Timi007) [inline addons/armory/ui/RscAttributes.hpp:1079]: This should be a function.

File: addons/armory/initSettings.hpp [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    "LIST",
+    [LLSTRING(cba_settings_actions), LLSTRING(cba_settings_actions_tooltip)],
+    [LELSTRING(main,category), LLSTRING(subCategory)],
+    [["ace_interaction","scroll_menu"],[LLSTRING(cba_settings_actions_ace_interaction), LLSTRING(cba_settings_actions_scroll_menu)],0],
+    1,
+    {}
+] call CBA_settings_fnc_init;
+
+[
+    QGVAR(cba_settings_playerDBConnection),
```
Comment (Timi007) [inline addons/armory/initSettings.hpp:12]:
```suggestion
    QGVAR(allowPlayerDBConnection),
```

File: addons/armory/initSettings.hpp [hunk 10]
```diff
@@ -0,0 +1,21 @@
+[
+    QGVAR(cba_settings_actions),
```
Comment (Timi007) [inline addons/armory/initSettings.hpp:2]:
```suggestion
    QGVAR(openUIActionMenu),
```

**Review by Timi007 — CHANGES_REQUESTED** [review]

File: addons/armory/stringtable.xml (hunk 4 above)
Comment (PhILoXs) [reply addons/armory/stringtable.xml]: STR_mts_Armory_displayName is now in use

File: addons/armory/stringtable.xml (hunk 6 above)
Comment (PhILoXs) [reply addons/armory/stringtable.xml]: Corrected

File: addons/armory/ui/RscAttributes.hpp (hunk 8 above)
Comment (PhILoXs) [reply addons/armory/ui/RscAttributes.hpp]: moved to function buttonCreateEquipment

File: addons/armory/ui/RscAttributes.hpp (hunk 7 above)
Comment (PhILoXs) [reply addons/armory/ui/RscAttributes.hpp]: moved to function openNewEquipmentDialog

File: addons/armory/initSettings.hpp (hunk 9 above)
Comment (PhILoXs) [reply addons/armory/initSettings.hpp]: Var is renamed to allowPlayerDBConnection

File: addons/armory/initSettings.hpp (hunk 10 above)
Comment (PhILoXs) [reply addons/armory/initSettings.hpp]: Var is renamed to openUIActionMenu

File: addons/armory/stringtable.xml (hunk 5 above)
Comment (PhILoXs) [reply addons/armory/stringtable.xml]: Not possible. The game uses the next correct stringtable entry if <br \> is used.

**Review by Timi007 — APPROVED** [review]

**Pull request merged**